### ISSUE TITLE
Function name after returns declaration not documented

### ISSUE THREAD

**Issue by duaraghav8** [ethereum/solidity#3423]
## Version
0.4.19

## Code
```
pragma solidity ^0.4.19;

contract Bar {
    function(address) external returns (address) foo;
}
```

## Platform
Remix Browser (uses solc)

## How to reproduce issue
Just type the above code into remix or compile with Solc. The code compiles fine. But it is not documented anywhere, including under [abstract contracts](http://solidity.readthedocs.io/en/develop/contracts.html#abstract-contracts) (notice that the name of the function - `foo` - is at the **end**). This syntax shouldn't be legal according to [official solidity grammar](http://solidity.readthedocs.io/en/develop/miscellaneous.html#language-grammar) (see `FunctionDefinition`)

## Problem
I am confused whether this syntax is intended or not. Can you please clarify its status

This problem originated from https://github.com/duaraghav8/Solium/issues/120
I couldn't find any issues/docs/SO discussions talking about this problem, apologies if it has already been discussed.

**Comment by chriseth** [comment]
Thanks for this issue! The code is in fact valid, it declares a variable of [function type](https://solidity.readthedocs.io/en/develop/types.html#function-types). I agree that it might be confusing, so if you want to add a comment to the part of the documentation that talks about function declarations and cross-link function types, that would be awesome!

**Comment by duaraghav8** [comment]
Ah! I mis-understood this syntax as a function declaration. Rather, its a variable declaration (a variable of type `function`). I'll send in a PR to docs for this & then close this issue.

**Comment by axic** [comment]
Fixed in #3423.

**Comment by chriseth** [comment]
@axic recursive comment?
